packages/client/src/proxy-action/ProxyAction.ts

```diff
@@ -83,4 +85,21 @@ export class ProxyAction {
       return result.data.result;
     });
   }
+
+  async waitForStatusComplete(
+    proxyActionId: string,
+  ): Promise<
+    | Result<
+        ProxyActionStatusResultFragment | ProxyActionErrorFragment | ProxyActionQueuedFragment,
+        CredentialsExpiredError | NotAuthenticatedError
+      >
+    | undefined
+  > {
+    return poll({
+      fn: () => this.checkStatus(proxyActionId),
+      validate: (result: Awaited<ReturnType<typeof this.checkStatus>>) => {
+        return getIsStatusCompleteFromProxyActionStatus(result.unwrap());
```

Comment:
`unwrap` would cause to throw the underling error, is `poll` then catching it and piping it back in the returned result?

Reply:
I improved that and a few other issues around polling in this PR https://github.com/lens-protocol/lens-sdk/pull/214